File: android/app/src/main/java/com/lightningkite/kwifttemplate/shared/views/DateRangeDemoVG.kt
package com.lightningkite.kwifttemplate.shared.views

import android.view.View
import com.lightningkite.kwift.observables.actual.bind
import com.lightningkite.kwift.observables.shared.StandardObservableProperty
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwift.views.shared.ViewGenerator
import com.lightningkite.kwifttemplate.layouts.DateRangeDemoXml
import java.util.*

class DateRangeDemoVG() : ViewGenerator() {
    override val title: String get() = "Date Range Demo"

    val start: StandardObservableProperty<Date?> = StandardObservableProperty(Date())
    val endInclusive: StandardObservableProperty<Date?> = StandardObservableProperty(Date())

    override fun generate(dependency: ViewDependency): View {
        val xml = DateRangeDemoXml()
        val view = xml.setup(dependency)

        xml.range.bind(start, endInclusive)
        xml.select.bind(start)

        return view
    }
}

File: iosOld/com/lightningkite/kwifttemplate/shared/views/DateRangeDemoVG.swift
//Package: com.lightningkite.kwifttemplate.shared.views
//Converted using Kwift2

import Foundation



public class DateRangeDemoVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Date Range Demo"
        }
    }
    public var start: StandardObservableProperty<Date?>
    public var endInclusive: StandardObservableProperty<Date?>
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = DateRangeDemoXml()
        var view = xml.setup(dependency)
        xml.range.bind(start, endInclusive)
        xml.select.bind(start)
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        self.start = StandardObservableProperty(Date())
        self.endInclusive = StandardObservableProperty(Date())
        super.init()
    }
}
 
